src/transports/http.c

```diff
@@ -349,10 +350,19 @@ static int on_header_value(http_parser *parser, const char *str, size_t len)
 	return 0;
 }
 
+GIT_INLINE(void) free_cred(git_cred **cred)
```

Comment:
I'd be tempted to move that as a function in the cred system. Also, I remember wondering as to why this specific struct happens to be open, but it's unrelated.

Reply:
Yeah, I thought about that - there's a `cred->free` function, but this is really concerned with the freeing of the `git_cred *` itself which is rather unique to the transport itself.

Reply:
Actually, there's already a `git_cred_free` function there, but I'm not sure if it matches your semantics (or even if you've switched to it now, as the diff is out of date).

Reply:
Aha - yes - I see what you're saying; that seems reasonable.